lib/rules-generator.js

```diff
@@ -405,24 +472,37 @@ util.methods(Generator, {
         if (replaceExp()) {
           exp.ref = ref;
           exp.args = args;
+          if (exp.ref.valueType == 'Snapshot') {
+            exp.valueType = 'Snapshot';
+          }
         }
       }
-      break;
-    }
 
-    return exp;
+      return exp;
+
+    default:
+      return exp;
```

Comment:
Will add comment - it's for expressions that don't need mutation to "expand" them (e.g. literals, etc)